packages/dashboard/src/components/task-summary-panel.tsx

```diff
@@ -0,0 +1,221 @@
+import React from 'react';
+import { SpotlightValue } from './spotlight-value';
+import Debug from 'debug';
+import * as RomiCore from '@osrf/romi-js-core-interfaces';
+import { IconButton, makeStyles, Typography } from '@material-ui/core';
+import { TreeItem, TreeView } from '@material-ui/lab';
+import ExpandMoreIcon from '@material-ui/icons/ExpandMore';
+import ChevronRightIcon from '@material-ui/icons/ChevronRight';
+import { TaskSummaryPanelItem } from './task-summary-panel-item';
+import PlayCircleFilledWhiteIcon from '@material-ui/icons/PlayCircleFilledWhite';
+import TaskManager from '../managers/task-manager';
+import { colorPalette } from '../util/css-utils';
+import { TreeButtonGroup } from './tree-button-group';
+import { Theme } from '@material-ui/core/styles/createMuiTheme';
+
+const debug = Debug('OmniPanel:TaskSummaryPanel');
+
+export interface TaskSummaryPanelProps {
+  allTasks: RomiCore.TaskSummary[];
+  spotlight?: Readonly<SpotlightValue<string>>;
+}
+
+export const TaskSummaryPanel = React.memo((props: TaskSummaryPanelProps) => {
+  debug('task summary status panel render');
+
+  const { allTasks, spotlight } = props;
+  const classes = useStyles();
+
+  React.useEffect(() => {
+    if (!spotlight) {
+      return;
+    }
+    // TODO: spotlight
+  }, [spotlight]);
+
+  const [taskContents, setTaskContents] = React.useState<{
+    [key: string]: JSX.Element;
+  }>({});
+  const [expanded, setExpanded] = React.useState<string[]>([]);
+  const [selected, setSelected] = React.useState<string>('');
+
+  // We need to persist across the renders
+  const savedTasksContent = React.useRef<{
+    [key: string]: JSX.Element;
+  }>({});
+
+  const handleToggle = (event: React.ChangeEvent<{}>, nodeIds: string[]) => {
+    setExpanded(nodeIds);
+  };
+
+  const handleSelect = (event: React.ChangeEvent<{}>, nodeIds: string) => {
+    setSelected(nodeIds);
+  };
+
+  const handleResetTasks = () => {
+    setExpanded([]);
+    setSelected('');
+  };
+
+  const handleClearAllCurrTasks = () => {
+    savedTasksContent.current = taskContents;
+    setTaskContents({});
+  };
+
+  const handleRestoreTasks = () => {
+    // Assigning another reference
+    let taskContentsTemp = Object.assign({}, taskContents);
+    Object.keys(savedTasksContent.current).forEach((element) => {
+      if (!(element in taskContents)) {
+        taskContentsTemp[element] = savedTasksContent.current[element];
+      }
+    });
+    setTaskContents(taskContentsTemp);
```

Comment:
You should be able to use the functional form to avoid recursive dependencies and use of a temp object.
```tsx
setTaskContents((currentContents) => {
  return doStuff(currentContents);
});
```